src/Cipher.ts

```diff
@@ -305,37 +289,19 @@ export function createDecipheriv(
 export function createCipher(
   algorithm: string,
   password: BinaryLike,
-  options?: Stream.TransformOptions
+  options?: Record<string, unknown> | Stream.TransformOptions
 ): Cipher {
   return new Cipher(algorithm, password, options);
 }
 
-// TODO(osp) on all the createCipheriv methods, node seems to use a "KeyObject" is seems to be a thread safe
-// object that creates keys and what not. Not sure if we should support it.
-// Fow now I replaced all of them to BinaryLike
-// export function createCipheriv(
-//   algorithm: CipherCCMTypes,
-//   key: BinaryLike,
-//   iv: BinaryLike,
-//   options: CipherCCMOptions
-// ): Cipher;
-// export function createCipheriv(
-//   algorithm: CipherOCBTypes,
-//   key: BinaryLike,
-//   iv: BinaryLike,
-//   options: CipherOCBOptions
-// ): CipherOCB;
-// export function createCipheriv(
-//   algorithm: CipherGCMTypes,
-//   key: BinaryLike,
-//   iv: BinaryLike,
-//   options?: CipherGCMOptions
-// ): Cipher;
+// export function createCipheriv(algorithm: CipherCCMTypes, key: CipherKey, iv: BinaryLike, options: CipherCCMOptions): CipherCCM;
+// export function createCipheriv(algorithm: CipherOCBTypes, key: CipherKey, iv: BinaryLike, options: CipherOCBOptions): CipherOCB;
+// export function createCipheriv(algorithm: CipherGCMTypes, key: CipherKey, iv: BinaryLike, options?: CipherGCMOptions): CipherGCM;
 export function createCipheriv(
   algorithm: string,
   key: BinaryLike,
   iv: BinaryLike | null,
-  options?: Stream.TransformOptions
+  options?: Record<string, unknown> | Stream.TransformOptions
```

Comment:
and here

Reply:
https://github.com/margelo/react-native-quick-crypto/pull/245/files#r1548175650